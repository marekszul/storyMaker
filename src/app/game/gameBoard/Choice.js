import React, { Component } from 'react'

class Choice extends Component {
    render () {
        return (
            <div><button onClick={() => this.props.onClick()}>Option {this.props.id}</button>{this.props.description}</div>
        )
    }
}

export default Choice