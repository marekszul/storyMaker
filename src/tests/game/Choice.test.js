import React from 'react'
import renderer from 'react-test-renderer'
import Choice from '../../app/game/gameBoard/Choice.js';
import { shallow,configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16'

configure({adapter: new Adapter()});
describe("Choice component tests", () =>
 {
    test("Render is ok", () => {
        let component = renderer.create(<Choice id="1" description="test description" onClick={() => {}}/>)
        let tree = component.toJSON();
        expect(tree).toMatchSnapshot();
     }) 
     test("Choice click should works", () => {
        let x = 0
        let component = shallow(<Choice id="2" description="test description2" onClick={() => x = 1}/>)
        component.find("button").simulate('click')
        expect(x).toBe(1);
     }) 
 })