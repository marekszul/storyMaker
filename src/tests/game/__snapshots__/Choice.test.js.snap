// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Choice component tests Render is ok 1`] = `
<div>
  <button
    onClick={[Function]}
  >
    Option 
    1
  </button>
  test description
</div>
`;
